WcaOnRails/app/models/user.rb

```diff
@@ -684,9 +684,7 @@ def can_edit_team?(team)
     can_manage_teams? ||
       team_leader?(team) ||
       # The leader of the WDC can edit the banned competitors list
-      (team == Team.banned && team_leader?(Team.wdc)) ||
-      # Senior Delegates and WFC Leader and Senior Members can edit Delegates on probation
-      (team == Team.probation && (senior_delegate? || team_leader?(Team.wfc) || team_senior_member?(Team.wfc)))
```

Comment:
Why is this check no longer necessary? The general statement from the comment directly above (`Senior Delegates and WFC Leader and Senior Members can edit Delegates on probation`) is still valid, even in your new `roles` system. So why is this check disappearing?